tests/ci/run_cryptofuzz.sh

```diff
@@ -11,50 +11,56 @@ set -u
 mkdir -p "$BUILD_ROOT"
-cd "$BUILD_ROOT"
 
-# Build AWS-LC based on https://github.com/guidovranken/cryptofuzz/blob/master/docs/openssl.md
-${CMAKE_COMMAND} -DCMAKE_CXX_FLAGS="$CXXFLAGS" -DCMAKE_C_FLAGS="$CFLAGS" -DBORINGSSL_ALLOW_CXX_RUNTIME=1 \
-  -GNinja -DBUILD_TESTING=OFF -DBUILD_LIBSSL=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo ../
-ninja
-cd ../
+# Build ACCP, but skip the tests since that's covered by all the other CI
+cd ${SRC_ROOT}
+./gradlew -x check build
+AWSLC_INSTALL_PATH=`realpath build/awslc/bin/`
+export AWSLC_INCLUDE_PATH="${AWSLC_INSTALL_PATH}/include/"
+export AWSLC_LIBRARY_PATH="${AWSLC_INSTALL_PATH}/lib64/"
+export CXXFLAGS="$CXXFLAGS -I $AWSLC_INCLUDE_PATH"
+export ACCP_JAR=`realpath build/lib/AmazonCorrettoCryptoProvider.jar`
 
-# CRYPTOFUZZ_AWSLC will enable AES_256_XTS in upstream code. See `V610453347`.
-# TODO: remove CRYPTOFUZZ_BORINGSSL when CRYPTOFUZZ_AWSLC is fully integrated.
-export CXXFLAGS="$CXXFLAGS -DCRYPTOFUZZ_BORINGSSL -DCRYPTOFUZZ_AWSLC"
-export OPENSSL_INCLUDE_PATH=`realpath include/`
-export OPENSSL_LIBCRYPTO_A_PATH=`realpath ${BUILD_ROOT}/crypto/libcrypto.a`
+# We run the Java module with ACCP as the JCE provider.
+# This module uses OpenSSL symbols, so we provide those from AWS-LC to avoid symbol conflicts
+# Java https://github.com/guidovranken/cryptofuzz/blob/master/docs/java.md
+export JDK_PATH=${JAVA_HOME}
+export LINK_FLAGS="${LINK_FLAGS} -L${JDK_PATH}/lib/server/ -ljvm -L${AWSLC_LIBRARY_PATH} -lcrypto"
+export LINK_FLAGS="${LINK_FLAGS} -Wl,-rpath=${JDK_PATH}/lib/server/:${AWSLC_LIBRARY_PATH}"
+export CXXFLAGS="$CXXFLAGS -DCRYPTOFUZZ_JAVA -DJAVA_WITH_ECDSA -DCRYPTOFUZZ_ACCP"
+cd "${CRYPTOFUZZ_SRC}/modules/java/"
+make -j$(nproc)
+
+# LSAN suppressions due to false positives
+cd ${FUZZ_ROOT}
+echo 'leak:libjvm.so' >> lsan.supp
+echo 'leak:libz.so' >> lsan.supp
+echo 'leak:libzip.so' >> lsan.supp
+export LSAN_OPTIONS="suppressions=`realpath lsan.supp`"
 
 # For cryptofuzz development only, override CRYPTOFUZZ_SRC with CUSTOM_CRYPTOFUZZ_REPO_DIR.
 CUSTOM_CRYPTOFUZZ_REPO_DIR=''
 if [[ -z "${CUSTOM_CRYPTOFUZZ_REPO_DIR}" ]]; then
   echo "CUSTOM_CRYPTOFUZZ_REPO_DIR is empty."
 else
   export CRYPTOFUZZ_SRC="${CUSTOM_CRYPTOFUZZ_REPO_DIR}"
-  # Local development does not need differential fuzzing.
-  # Remove these libs related build flags.
-  export CXXFLAGS="${CXXFLAGS// -DCRYPTOFUZZ_BOTAN/}"
-  export CXXFLAGS="${CXXFLAGS// -DCRYPTOFUZZ_CRYPTOPP/}"
   cd "$CRYPTOFUZZ_SRC"
   # This step is to generate required header and cpp files.
   python3 gen_repository.py
 fi
 
-# Build the common OpenSSL module with AWS-LC
-cd "${CRYPTOFUZZ_SRC}/modules/openssl"
-make "-j${NUM_CPU_THREADS}"
-
-export CFLAGS="${CFLAGS} -I $OPENSSL_INCLUDE_PATH"
-export CXXFLAGS="${CXXFLAGS} -I $OPENSSL_INCLUDE_PATH"
+# Required linker flag set via this variable
 export LIBFUZZER_LINK="-fsanitize=fuzzer"
+# Explicitly disable OpenSSL module since it creates symbol conflicts
+export CXXFLAGS="$CXXFLAGS -DCRYPTOFUZZ_NO_OPENSSL"
```

Comment:
is the implication here that we won't be able to do differential fuzzing against OpenSSL until we statically link AWS-LC to libaccp?

Reply:
Yes, but once we statically link AWS-LC it becomes a lot more clear cut how ld ties the symbols at run time and that's a straightforward PR. That said, we won't be able to reliably differential test against OpenSSL because of the symbol conflicts. If we bang at it for a lot longer, we can probably get it to work but I do question the value prop when the simple thing is to do is to just differential fuzz against something that has none of these problems.

Reply:
> I do question the value prop when the simple thing is to do is to just differential fuzz against something that has none of these problems.

good point, and differential fuzzing against OpenSSL would be a little redundant if AWS-LC already does that. i suppose differential fuzzing would only be useful for us for compatibility checks against other java providers, but we've already discussed the limtations around that.